systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomKeyPairST.java

```diff
@@ -0,0 +1,291 @@
+     * @throws NoSuchAlgorithmException if specified algorithm for truststore is not supported
+     * @throws IOException              if an input or output file could not be read/written.
+     * @throws InvalidKeySpecException  if any problems with reading/writing the truststore
+     */
+    void helperMethod(final ExtensionContext extensionContext, final BiFunction<TestStorage, SystemTestCertHolder, SystemTestCertHolder> publicKeyPart,
+                      final BiFunction<TestStorage, SystemTestCertHolder, Void> privateKeyPart,
+                      final Function<TestStorage, Void> rollingComponents) throws NoSuchAlgorithmException, IOException, InvalidKeySpecException {
+        final TestStorage ts = new TestStorage(extensionContext);
+
+        // 0. Generate root and intermediate certificate authority
+        SystemTestCertHolder stCertHolder = new SystemTestCertHolder(extensionContext);
+
+        // 1. Prepare correspondent Secrets from generated custom CA certificates
+        stCertHolder.prepareSecretsFromBundles(ts.getNamespaceName(), ts.getClusterName());
+
+        // 2. Create a Kafka cluster without implicit generation of CA
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(ts.getClusterName(), 1)
+            .editOrNewSpec()
+                .withNewClientsCa()
+                    .withRenewalDays(5)
+                    .withValidityDays(20)
+                    .withGenerateCertificateAuthority(false)
+                .endClientsCa()
+            .endSpec()
+            .build());
+
+        // 3. Pause the reconciliation of the Kafka custom resource
+        LOGGER.info("Pause the reconciliation of the Kafka custom resource ({}).", KafkaResources.kafkaStatefulSetName(ts.getClusterName()));
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(ts.getClusterName(), kafka -> {
+            Map<String, String> kafkaAnnotations = kafka.getMetadata().getAnnotations();
+            if (kafkaAnnotations == null) {
+                kafkaAnnotations = new HashMap<>();
+            }
+            // adding pause annotation
+            kafkaAnnotations.put(Annotations.ANNO_STRIMZI_IO_PAUSE_RECONCILIATION, "true");
+            kafka.getMetadata().setAnnotations(kafkaAnnotations);
+        }, ts.getNamespaceName());
+
+        // 4. Check that the status conditions of the custom resource show a change to ReconciliationPaused
+        KafkaUtils.waitForKafkaStatus(ts.getNamespaceName(), ts.getClusterName(), CustomResourceStatus.ReconciliationPaused);
+
+        // 5. Update the Secret for the CA certificate. (public key part)
+        stCertHolder = publicKeyPart.apply(ts, stCertHolder);
+
+        // 6. Update the Secret for the CA key used to sign your new CA certificate. (private key part)
+        privateKeyPart.apply(ts, stCertHolder);
+
+        // 8. save the current state of the Kafka, ZooKeeper and EntityOperator pods
+        // 9. Resume reconciliation from the pause.
```

Comment:
Where do you unpause it?

Reply:
He unpauses it in a test. 
I was looking for pausing annotation at my first check of this PR. :-)